### PR TITLE
E2E test pipeline download and install old version of LTW and CP

### PR DESCRIPTION
**Pull request by cikelengfeng**
In order to support update scenarios, we need to download old version of LTW and CompanyPortal.

- Changed "broker-test.yml" to pass old version name of LTW and CP to "download-brokers-and-azure-sample.yml".
- Changed "broker-test.yml" to push old version apks of LTW and CP to firebase devices.
- Changed "download-brokers-and-azure-sample.yml" to consume old version name of LTW and CP to download them to brokerapks/oldAPKs directory.

### CODE REVIEW

File: azure-pipelines/ui-automation/broker-test.yml [hunk 1]
```diff
@@ -83,6 +85,10 @@ parameters:
   displayName: Company Portal Version
   type: string
   default: '*'
+- name: oldCompanyPortalVersion
+  displayName: Old Company Portal Version
+  type: string
+  default: '5.0.5729'
```
Comment (somalaya) [inline azure-pipelines/ui-automation/broker-test.yml:91]: Have we already pushed these versions to the feed?

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml [hunk 2]
```diff
@@ -95,6 +111,19 @@ jobs:
     - script: mv ./YPC-*.apk ./LTW-signed.apk
       displayName: 'Rename LTW build to LTW-signed.apk'
       workingDirectory: '$(Build.ArtifactStagingDirectory)/brokers'
+    - task: UniversalPackages@0
+      displayName: 'Download old com.microsoft.appmanager'
```
Comment (somalaya) [inline azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:115]: How about display name being Download old LTW apk?

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml [hunk 3]
```diff
@@ -95,6 +111,19 @@ jobs:
...
+        command: 'download'
+        downloadDirectory: '$(Build.ArtifactStagingDirectory)/brokers/oldAPKs'
+        feedsToUse: 'external'
+        externalFeedCredentials: '${{ parameters.OSServiceConnection }}'
+        feedDownloadExternal: '${{ parameters.LTWFeedName }}'
+        packageDownloadExternal: 'com.microsoft.appmanager'
+        versionDownloadExternal: '${{ parameters.OldLTWVersion }}'
+    - script: mv ./YPC-*.apk ./OldLTW-signed.apk
+      displayName: 'Rename Old LTW build to OldLTW-signed.apk'
+      workingDirectory: '$(Build.ArtifactStagingDirectory)/brokers/oldAPKs'
```
Comment (somalaya) [inline azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:126]: can you help me understand why renaming the apk is needed?

File: azure-pipelines/ui-automation/broker-test.yml (hunk 1 above)
Comment (cikelengfeng) [reply azure-pipelines/ui-automation/broker-test.yml:91]: Yes, I randomly picked it from here, we can easily change this version in the future. https://msazure.visualstudio.com/One/_artifacts/feed/Android-Broker@Local/UPack/com.microsoft.windowsintune.companyportal-signed/overview/5.0.5729

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml (hunk 3 above)
Comment (cikelengfeng) [reply azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:126]: I saw we download the latest LTW apk and rename it to LTW-signed.apk in the existing pipeline, so I think it's better to follow the existing pattern.

I guess the original idea of doing this is to get a shorter and meaningful file name?

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml (hunk 2 above)
Comment (cikelengfeng) [reply azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml]: Changed to "Download Old LTW Apk", and also changed displayName for CompanyPortal

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml [hunk 4]
```diff
@@ -16,6 +16,9 @@ parameters:
   - name: companyPortalVersion
     displayName: Company Portal Version
     type: string
+  - name: oldCompanyPortalVersion
+    displayName: Old Company Portal Version
+    type: string
```
Comment (fadidurah) [inline azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:21]: We should have a default here, or else we will break other pipelines using this step

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml [hunk 5]
```diff
@@ -33,6 +36,9 @@ parameters:
     displayName: Link to Windows Version
     type: string
     default: '*'
+  - name: oldLTWVersion
+    displayName: Old Link to Windows Version
+    type: string
```
Comment (fadidurah) [inline azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:41]: Same here

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml (hunk 4 above)
Comment (fadidurah) [reply azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:21]: Let's use the versions you defined in broker-test.yml

Comment (cikelengfeng) [reply azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:22]: Done.

File: azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml (hunk 5 above)
Comment (cikelengfeng) [reply azure-pipelines/ui-automation/templates/download-brokers-and-azure-sample.yml:43]: Done.

**Review by somalaya — APPROVED** [review]

**Review by fadidurah — APPROVED** [review]

**Pull request merged**